src/class/HTTPRequest/HTTPRequest.cpp

```diff
@@ -0,0 +1,162 @@
+#include <HTTPRequest/HTTPRequest.hpp>
+
+/*
+** ------------------------------- CONSTRUCTOR --------------------------------
+*/
+
+const int HTTPRequest::MAX_CHUNK_ATTEMPTS = 5;
+
+HTTPRequest::HTTPRequest(Server *virtualServer, int clientFd)
+	: _virtualServer(virtualServer)
+	, _clientFd(clientFd)
+	, _requestAttempts(0)
+	, _responseAttempts(0)
+	, _bytesSent(0)
+	, _response("\0")
+	, _basicRequest("")
+{
+}
+
+const char *HTTPRequest::SendingResponseError::what() const throw()
+{
+	return "Couldn't send response: send() failed";
+}
+
+const char *HTTPRequest::RecievingRequestError::what() const throw()
+{
+	return "Didn't recieve request: recv() failed";
+}
+
+/*
+** -------------------------------- DESTRUCTOR --------------------------------
+*/
+
+HTTPRequest::~HTTPRequest()
+{
+}
+
+/*
+** --------------------------------- OVERLOAD ---------------------------------
+*/
+
+std::ostream &operator<<(std::ostream &o, HTTPRequest const &i)
+{
+	o << "HTTPRequest[" << i.getBasicRequest() << "]";
+	return o;
+}
+
+/*
+** --------------------------------- METHODS ----------------------------------
+*/
+
+void HTTPRequest::recvRequest(void)
+{
+	int tmpRecvLen;
+	char tmpRaw[4096];
+
+	tmpRecvLen = recv(this->_clientFd, tmpRaw, sizeof(tmpRaw), MSG_DONTWAIT);
+	this->_requestAttempts++;
+	this->_rawRequest += tmpRaw;
+	if (tmpRecvLen <= -1)
```

Comment:
i think there is a case of tmpRecvLen == 0 where the client closed the connection.
it is not an error, and we will need to remove the client from poll